### PR TITLE
Update modal styles

### LINKED ISSUE #10661: Modals: Alignment and scroll artifact bugs
**Setup**
- Rancher version: v2.9-head
- Rancher UI Extensions: Elemental
- Browser type & version:
  - Brave Version 1.61.104 Chromium: 120.0.6099.115 (Official Build)  (64-bit)
  - Chrome Version 123.0.6312.58 (Official Build) (64-bit)

**Describe the bug**
Bug 1
- Modal title is not vertically aligned
Bug 2
- On linux brave and chrome scroll artifacts are shown (white blank square bottom right in brave, disabled scroll bars in chrome)

**To Reproduce**
- Nav to Custom Cluster Badge modal from explorer cluster dashboard OR delete a resource with a warning modal

**Result**
- As per screenshots

**Expected Result**
- title aligned in middle of vertical space
- no scroll artifacts

**Screenshots**
Brave
image

image

Chrome
image

image

**Additional context**
- artefacts caused by `.modal-overlay .modal-container` `    overflow: scroll;`. see https://github.com/rancher/dashboard/pull/10556/files#diff-ffbc7799167852e06e43d04fe1a43795f231a53841ec183530113103493a0e96R94
- also tested on windows chrome, and see the same as above

### PR DESCRIPTION
**Pull request by rak-phillip**
### Summary
This addresses a few issues with the `AppModal` component that were detailed in #10661.

Fixes #10661

### Occurred changes and/or fixed issues

- Fixes an error with the width validator when a valid string is provided
- Specifies `overflow: auto` on the modal container so that scrollbars are not present in chrome/brave
- Adjusts the bottom margin of the title in `shell/dialog/AddCustomBadgeDialog.vue` so that it appears as properly centered

### Technical notes summary

Centering of the title was not a regression introduced in #9537, but is a result of the design of the card component. More details can be found here https://github.com/rancher/dashboard/issues/10661#issuecomment-2010031758

### Areas or cases that should be tested

See the issue for more info.

### Screenshot/Video

centered-title

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

File: shell/components/__tests__/AppModal.test.ts [hunk 1]
```diff
@@ -67,4 +67,36 @@ describe('appModal', () => {
...
+
+    await wrapper.setProps({ width: '600px' });
+    const container = wrapper.find('.modal-container');
+
+    expect(container.exists()).toBeTruthy();
+    expect(wrapper.find('.modal-container').element.style.width).toBe('600px');
+
+    expect(consoleErrorSpy).not.toHaveBeenCalled();
+
+    consoleErrorSpy.mockRestore();
```
[the remark this answers is not in the document]
Comment (rak-phillip) [reply shell/components/__tests__/AppModal.test.ts:84]: I think it's sufficient.

File: shell/components/__tests__/AppModal.test.ts [hunk 2]
```diff
@@ -67,4 +67,36 @@ describe('appModal', () => {
...
+  it('generates validation errors with an invalid sets a string for width', async() => {
+    const consoleErrorSpy = jest.spyOn(console, 'error');
+
+    consoleErrorSpy.mockImplementation(() => {});
+
+    await wrapper.setProps({ width: 'FAIL' });
+    const container = wrapper.find('.modal-container');
+
+    expect(container.exists()).toBeTruthy();
+    expect(wrapper.find('.modal-container').element.style.width).toBe('600px');
```
[the remark this answers is not in the document]
Comment (rak-phillip) [reply shell/components/__tests__/AppModal.test.ts:96]: We're asserting that the modal renders with the default width in the event that an invalid value is provided for the `width` prop.

[the remark this answers is not in the document]
Comment (cnotv) [reply shell/components/__tests__/AppModal.test.ts:94]: yes but it's the same also when it gets an error

[the remark this answers is not in the document]
Comment (rak-phillip) [reply shell/components/__tests__/AppModal.test.ts:94]: I updated the case above to better communicate that we're expecting a different value on success. lmk if that improves the clarity of the tests

[the remark this answers is not in the document]
Comment (cnotv) [reply shell/components/__tests__/AppModal.test.ts:94]: yep, definitely more sense now 👍

**Review by cnotv — APPROVED** [review]
LGTM

**Pull request merged**